trogon/detect_run_string.py

```diff
@@ -0,0 +1,39 @@
+import os
+import shlex
+import sys
+
+
+def detect_run_string(path=None, _main=sys.modules["__main__"]):
+    """This is a slightly modified version of a function from Click.
+    """
+    if not path:
+        path = sys.argv[0]
+
+    # The value of __package__ indicates how Python was called. It may
+    # not exist if a setuptools script is installed as an egg. It may be
+    # set incorrectly for entry points created with pip on Windows.
+    if getattr(_main, "__package__", None) is None or (
+        os.name == "nt"
+        and _main.__package__ == ""
+        and not os.path.exists(path)
+        and os.path.exists(f"{path}.exe")
+    ):
+        # Executed a file, like "python app.py".
+        file_path = shlex.quote(os.path.basename(path))
+        if sys.orig_argv[0] == "python":
```

Comment:
This `sys.orig_argv` is only available for python 3.10+ (whereas the pyproject.toml stipulates python 3.7)

Reply:
Thanks - we have an alternative solution for this that doesn't used `orig_argv` :)